### PR TITLE
doc: Add patching vulnerabilities step on release doc

### PR DESCRIPTION
**Pull request by kavirajk**
**What this PR does / why we need it**:

Follow up to other release related docs #10646 https://github.com/grafana/loki/pull/10676

Goal is make the doc complete for all type of Loki releases. Stable/Patch/Security releases.

Changes:
1. Two new docs
     a. Patch-Vulnerabilities Overview - patching go dependencies, Go compiler and Docker images
     b. Patch-Go-Version - How to patch Go version on both loki build image and loki binaries.
2. Renamed files to remove number prefix on the file name e.g: `1-create-release-branch` to just `create-release-branch`. Makes it easy to add more docs without much change.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

Follow up to other release related docs #10646 https://github.com/grafana/loki/pull/10676

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added

### CODE REVIEW

File: docs/sources/community/maintaining/release/patch-go-version.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+title: Patch Go version
+description: Patch Go version
```
Comment (chaudum) [inline docs/sources/community/maintaining/release/patch-go-version.md:3]:
```suggestion
description: Describes the procedure how to patch the Go version in the Loki build image.
```

File: docs/sources/community/maintaining/release/patch-vulnerabilities.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+title: Patch Vulnerabilities
+description: Patch Vulnerabilities
```
Comment (chaudum) [inline docs/sources/community/maintaining/release/patch-vulnerabilities.md:3]:
```suggestion
description: Describes the procedure how to patch Loki to mitigate vulnerabilities.
```

File: docs/sources/community/maintaining/release/patch-vulnerabilities.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+1. Grafana Loki source code.
+
+1. Grafana Loki dependencies.
+
+Grafana Loki dependencies can be
+
+1. Go dependencies in `go.mod`
+
+1. Go compiler itself
```
Comment (chaudum) [inline docs/sources/community/maintaining/release/patch-vulnerabilities.md:23]:
```suggestion
1. the Go compiler itself
```

File: docs/sources/community/maintaining/release/patch-vulnerabilities.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+1. Grafana Loki docker dependencies (example: base images)
```
Comment (chaudum) [inline docs/sources/community/maintaining/release/patch-vulnerabilities.md:25]:
```suggestion
1. Grafana Loki Docker dependencies, for example, the base images
```

File: docs/sources/community/maintaining/release/patch-vulnerabilities.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	1. Check if [dependabot already patched the dependency](https://github.com/grafana/loki/pulls?q=is%3Apr+label%3Adependencies+is%3Aclosed) or [have a PR opened to patch](https://github.com/grafana/loki/pulls?q=is%3Apr+is%3Aopen+label%3Adependencies) . If not, manually upgrade the package on the `main` branch as follows.
+
+		```shell
+		go get -u -v <package-path>@<patched-version>
+		go mod tidy
+		go mod vendor
+		```
+	1. [Backport]({{< relref "./backport-commits" >}}) it to `release-$VERSION_PREFIX` branch.
+
+	1. Repeat for each Go dependencies
```
Comment (chaudum) [inline docs/sources/community/maintaining/release/patch-vulnerabilities.md:52]:
```suggestion
	1. Repeat for each Go dependency
```

File: docs/sources/community/maintaining/release/patch-vulnerabilities.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+1. [Patch Go compiler]({{< relref "./patch-go-version" >}}).
+
+1. Patch Grafana Loki docker dependencies(example: alpine images).
```
Comment (chaudum) [inline docs/sources/community/maintaining/release/patch-vulnerabilities.md:56]:
```suggestion
1. Patch Grafana Loki Docker dependencies, for example: Alpine Linux base images.
```

File: docs/sources/community/maintaining/release/patch-vulnerabilities.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+   1. Update docker image version. [Example PR](https://github.com/grafana/loki/pull/10573).
```
Comment (chaudum) [inline docs/sources/community/maintaining/release/patch-vulnerabilities.md:58]:
```suggestion
   1. Update Docker image version. [Example PR](https://github.com/grafana/loki/pull/10573).
```

**Review by chaudum — COMMENTED** [review]
Thanks, that documentation is really helpful!

File: docs/sources/community/maintaining/release/_index.md [hunk 8]
```diff
@@ -12,23 +12,32 @@ to publish a new [Grafana Loki](https://github.com/grafana/loki) release.
...
-1. [Create release branch]({{< relref "./1-create-release-branch" >}})
-1. [Backport commits]({{< relref "./2-backport-commits" >}})
-1. [Prepare Release notes]({{< relref "./3-prepare-release-notes" >}})
-1. [Prepare Changelog]({{< relref "./4-prepare-changelog" >}})
-1. [Document Metrics and Configurations changes]({{< relref "./5-document-metrics-configurations-changes" >}})
-1. [Prepare Upgrade guide]({{< relref "./6-prepare-upgrade-guide" >}})
-1. [Update version numbers]({{< relref "./7-update-version-numbers" >}})
-1. [Tag Release]({{< relref "./8-tag-release" >}})
-1. [Publish Release]({{< relref "./9-publish-release" >}})
+1. [Create release branch]({{< relref "./create-release-branch" >}})
```
Comment (poyzannur) [inline docs/sources/community/maintaining/release/_index.md:15]: When I click "View file" in the options above I expect the `relref`s to be rendered correctly as links, but they don't. Are we sure removing the numbers don't break it?

**Review by chaudum — APPROVED** [review]
lgtm

File: docs/sources/community/maintaining/release/_index.md (hunk 8 above)
Comment (kavirajk) [reply docs/sources/community/maintaining/release/_index.md:15]: I think `relref` is a hugo template that get rendered on grafana website. I don't think it renders correctly on Github AFAIK.

e.g: similar relref on other pages https://github.com/grafana/loki/blob/cd7c40f9eea535fdcba1428af617b8e303085c6d/docs/sources/get-started/_index.md

**Comment by JStickler** [comment]
You wrote up a new topic, `patch-go-versions`, but it doesn't look like it was linked to the README.

**Pull request merged**